simul/platform/deterlab.go

```diff
@@ -345,17 +346,70 @@ func (d *Deterlab) Wait() error {
 func (d *Deterlab) createHosts() {
 	numServers := d.Servers
 
-	ip := "10.255.0."
-	name := d.Project + ".isi.deterlab.net"
+	// Query deterlab's API for servers
+	log.Lvl2("Querying Deterlab's API to retrieve server names and addresses")
+	command := fmt.Sprintf("/usr/testbed/bin/expinfo -l -e %s,%s", d.Project, d.Experiment)
+	cmdOutputBytes, err := SSHRun(d.Login, d.Host, command)
+	if err != nil {
+		log.Fatal("Error while querying Deterlab:", err)
+	}
+
+	// Parse the output
+	outputStr := string(cmdOutputBytes)
+
+	// extract the interesting bit
+	str1 := strings.Split(outputStr, "Link Info:\n")
+	str2 := strings.Split(str1[1], "\n\n")
+	str3 := strings.Split(str2[0], "---------\n")
+	interestingBit := str3[1]
+	lines := strings.Split(interestingBit, "\n")
+
+	// extract one line over two
+	interestingLines := make([]string, 0)
+	for i, line := range lines {
+		if i%2 == 0 {
+			interestingLines = append(interestingLines, line)
+		}
+	}
+
+	// format : LAN	name:interface	ip	[otherdata]
+	// number of spaces is varying
+	regex := regexp.MustCompile("^([^ ]+)([ ]+)([^ ]+)([ ]+)([^ ]+)")
+	// lost with this regex ? http://regexr.com/3fe0m
+
 	d.Phys = make([]string, 0, numServers)
 	d.Virt = make([]string, 0, numServers)
-	for i := 1; i <= numServers; i++ {
-		d.Phys = append(d.Phys, fmt.Sprintf("server-%d.%s.%s", i-1, d.Experiment, name))
-		d.Virt = append(d.Virt, fmt.Sprintf("%s%d", ip, i))
+
+	for i := range interestingLines {
+		matches := regex.FindStringSubmatch(interestingLines[i])
+
+		if len(matches) < 6 {
```

Comment:
No, the API yields other infos (more than 6). This is just a check for when the API will change in background.
We can remove this weird check and pray that the API does not change, or keep it and maybe be warned. What do you think ?